events/moose/moose.go

```diff
@@ -658,6 +687,222 @@ func (s *Subscriber) sendEvent(contentType, userAgent, requestBody string) int {
 	return errCodeEventSendSuccess
 }
 
+func (s *Subscriber) fetchSubscriptions() error {
+	if !s.enabled {
+		return nil
+	}
+	var cfg config.Config
+	if err := s.Config.Load(&cfg); err != nil {
+		return fmt.Errorf("loading config: %w", err)
+	}
+	token := cfg.TokensData[cfg.AutoConnectData.ID].Token
+
+	payments, err := s.SubscriptionAPI.Payments(token)
+	if err != nil {
+		return fmt.Errorf("fetching payments: %w", err)
+	}
+
+	orders, err := s.SubscriptionAPI.Orders(token)
+	if err != nil {
+		return fmt.Errorf("fetching orders: %w", err)
+	}
+
+	payment, ok := findPayment(payments)
+	if !ok {
+		return fmt.Errorf("no valid payments found for the user")
+	}
+
+	var orderErr error
+	order, ok := findOrder(payment, orders)
+	if !ok {
+		orderErr = fmt.Errorf("no valid order was found for the payment")
+	}
+
+	if err := s.setSubscriptions(
+		payment,
+		order,
+		countFunc(payments, isPaymentValid, 2),
+	); err != nil {
+		errors.Join(orderErr, fmt.Errorf("setting subscriptions: %w", err))
+	}
+
+	return orderErr
+}
+
+func findPayment(payments []core.PaymentResponse) (core.Payment, bool) {
+	// Sort by CreatedAt descending
+	slices.SortFunc(payments, func(a core.PaymentResponse, b core.PaymentResponse) int {
+		return -a.Payment.CreatedAt.Compare(b.Payment.CreatedAt)
+	})
+
+	// Find first element matching criteria
+	index := slices.IndexFunc(payments, isPaymentValid)
+	if index < 0 {
+		return core.Payment{}, false
+	}
+
+	return payments[index].Payment, true
+}
+
+func findOrder(p core.Payment, orders []core.Order) (core.Order, bool) {
+	// Find order matching the payment
+	if p.Subscription.MerchantID != 25 && p.Subscription.MerchantID != 3 {
```

Comment:
What do those numbers mean? :thinking: Maybe let's put it in consts?

Reply:
These are from the docs